examples/binding_affinity_prediction/main.py

```diff
@@ -23,80 +24,166 @@ def run_a_train_epoch(args, epoch, model, data_loader,
+                print(val_msg)
+                val_r2[epoch] = val_scores['r2']
+            if len(test_set) > 0:
+                test_scores = run_an_eval_epoch(args, model, test_loader)
+                test_msg = update_msg_from_scores('test results', test_scores)
+                print(test_msg)
+                test_r2[epoch] = test_scores['r2']
+            print('')
+        # save model r2 at each epoch
+        if args['save_r2']:
+            save_path = args['save_r2'] + "/{}_{}_{}_{}_trial{}.npz".format(args['model'], args['version'], args['subset'], args['split'], trial+1)
+            np.savez(save_path, train_r2=train_r2, val_r2=val_r2, test_r2=test_r2)
+
+        # save results on the epoch with best validation r2
+        best_epoch = np.argmax(val_r2)
+        trial_train_r2[trial] = train_r2[best_epoch]
+        trial_val_r2[trial] = val_r2[best_epoch]
+        trial_test_r2[trial] = test_r2[best_epoch]
+        print('Best test epoch: ', best_epoch + 1)
+
+    print(f'\n Finished {n_trials} trials.')
+    print('Best train R2 mean: ', trial_train_r2.mean(), ', standard deviation: ', trial_train_r2.std())
+    print('Best validation R2 mean: ', trial_val_r2.mean(), ', standard deviation: ', trial_val_r2.std())
+    print('Best test R2 mean: ', trial_test_r2.mean(), ', standard deviation: ', trial_test_r2.std())
 
-    test_scores = run_an_eval_epoch(args, model, test_loader)
-    test_msg = update_msg_from_scores('test results', test_scores)
-    print(test_msg)
 
 if __name__ == '__main__':
     import argparse
 
     from configure import get_exp_configure
 
     parser = argparse.ArgumentParser(description='Protein-Ligand Binding Affinity Prediction')
-    parser.add_argument('-m', '--model', type=str, choices=['ACNN'],
+    parser.add_argument('-m', '--model', type=str, choices=['ACNN', 'PotentialNet'],
                         help='Model to use')
     parser.add_argument('-d', '--dataset', type=str,
                         choices=['PDBBind_core_pocket_random', 'PDBBind_core_pocket_scaffold',
                                  'PDBBind_core_pocket_stratified', 'PDBBind_core_pocket_temporal',
                                  'PDBBind_refined_pocket_random', 'PDBBind_refined_pocket_scaffold',
-                                 'PDBBind_refined_pocket_stratified', 'PDBBind_refined_pocket_temporal'],
-                        help='Dataset to use')
+                                 'PDBBind_refined_pocket_stratified', 'PDBBind_refined_pocket_temporal',
+                                 'PDBBind_refined_pocket_structure', 'PDBBind_refined_pocket_sequence'],
+                        help='Data subset to use')
+    parser.add_argument('-v', '--version', type=str, choices=['v2007', 'v2015'], default='v2015')
+    parser.add_argument('--save_r2', type=str, default='', help='path (need to be existing directory) to save r2 at each epoch, default not save')
```

Comment:
We can create a folder even if the directory does not exist yet.

Reply:
Implemented